### PR TITLE
Key upload fixes and improvements

### PR DESCRIPTION
**Pull request by Kami**
Some fixes for issues I've noticed while debugging the "key pair doesn't match" issue.

1. Use a more safe default key size (2048 bits), 1024 bits key are considered unsafe today.
2. Use a securely randomly generated file path where the public and private key is stored. If the web server process runs with elevated privileges and it uses a fixed temp path this introduces a security issue (potential escalation via symlink...)
3. Clarify that we don't support password protected private keys, improve the error message.

Fixes: #51

### CODE REVIEW

File: st2installer/controllers/keypair.py [hunk 1]
```diff
@@ -57,4 +73,39 @@ def private(self):
   def public(self):
     response.headers['Content-Disposition'] = 'attachment; filename="st2-ssh.pub"'
     return self.gen_public.exportKey('OpenSSH')
-  
\ No newline at end of file
+
+  def _parse_query_params(self, request):
```
Comment (Kami) [inline st2installer/controllers/keypair.py]: This is proudly borrowed from st2.

**Comment by Kami** [comment]
I still need to update tests, but it's a bit hard since not all of them pass locally and on Travis on master (before this change).

Ideally, we would wait for tests on master to be fixed / changed and then merge this branch.

**Comment by emedvedev** [comment]
I made the tests pass. Merge master into this branch, update the tests and Travis should be good.

**Comment by Kami** [comment]
@emedvedev Great, thanks. I will have a look.

**Comment by Kami** [comment]
OK, tests pass. Merging.

**Pull request merged**